### PR TITLE
Fe 9 2 make dynamic uploader

### PR DESCRIPTION
**Pull request by salammohamadi**

### CODE REVIEW

File: src/components/dataPage/dataTable/DataTableColumn.tsx [hunk 1]
```diff
@@ -23,24 +23,25 @@ const DataTableColumn = <T, >(props: React.PropsWithChildren<IProps<T>>) => {
...
       return <p>{new Date(Date.parse(value[fieldKey])).toLocaleDateString('fa', { dateStyle: 'long' })}</p>;
 
     case InputType.CHECKBOX:
       return <p>{repairUnit.find((opt) => opt.value === value[fieldKey])?.label}</p>;
 
-    case InputType.UPLOADER:
-    { const [visible, setVisible] = React.useState(false);
+    case InputType.IMG_UPLOADER:
+    {
+      const [visible, setVisible] = React.useState(false);
```
Comment (da-saman) [inline src/components/dataPage/dataTable/DataTableColumn.tsx:36]: can we use react hooks like this? I think no. react hooks should not be used in conditional cases, dont know why react is not throwing an error for this

Comment (da-saman) [reply src/components/dataPage/dataTable/DataTableColumn.tsx:36]: I Think better to use a separate component for this

File: src/utils/schemas/equipmentSchema.ts [hunk 2]
```diff
@@ -1,9 +1,17 @@
 import * as yup from 'yup';
```
Comment (da-saman) [inline src/utils/schemas/equipmentSchema.ts:2]: can we mix  equipmentFormPart1Schema and equipmentFormPart2Schema here?

File: src/components/equipmentFormModal/EquipmentFilesForm.tsx [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        reset(Object(dataPageInfo.editingRecord));
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  React.useEffect(() => {
+    const equipmentData = dataPageInfo.editingRecord as EquipmentResponse;
+    const docsObject : EquipmentRequestPart2 = {
```
Comment (da-saman) [inline src/components/equipmentFormModal/EquipmentFilesForm.tsx:93]: better to create a function outside the component to extract those fields;
and use it here and in line 42

File: src/components/equipmentFormModal/EquipmentForm.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+const EquipmentForm = (props: IProps) => {
+  const {
+    onSetActiveKey,
+    onSetFormValues,
+  } = props;
+
+  const dataPageInfo = useSelector((state: StoreType) => state.dataPageInfo);
+
+  const equipmentPart1Data : EquipmentRequestPart1 = {
```
Comment (da-saman) [inline src/components/equipmentFormModal/EquipmentForm.tsx:28]: not used

File: src/utils/schemas/equipmentSchema.ts (hunk 2 above)
Comment (salammohamadi) [reply src/utils/schemas/equipmentSchema.ts:2]: export const equipmentSchema = equipmentFormPart1Schema.concat(equipmentFormPart2Schema);

File: src/components/equipmentFormModal/EquipmentForm.tsx (hunk 4 above)
Comment (salammohamadi) [reply src/components/equipmentFormModal/EquipmentForm.tsx:28]: The last two comments was written by you, so no idea what's going on. maybe we should do this part together.

**Review by da-saman — APPROVED** [review]

**Pull request merged**